hasktorch/src/Torch/Typed/Native.hs

```diff
@@ -2943,9 +2943,72 @@ qZeroPoint input = unsafePerformIO $ cast1 ATen.q_zero_point_t input
 -- combinations :: Tensor device dtype shape -> Int -> Bool -> Tensor device dtype shape
 -- combinations _input _r _with_replacement = unsafePerformIO $ (cast3 ATen.combinations_tlb) _input _r _with_replacement
 
--- | lstmCell
--- >>> dtype &&& shape $ fst $ lstmCell (ones :: CPUTensor 'D.Float '[12,2]) (ones :: CPUTensor 'D.Float '[12,3]) (ones :: CPUTensor 'D.Float '[12]) (ones :: CPUTensor 'D.Float '[12]) ((ones :: CPUTensor 'D.Float '[2,3]), (ones :: CPUTensor 'D.Float '[2,3])) (ones :: CPUTensor 'D.Float '[2,2])
--- (Float,[2,3])
+-- | lstm
+-- Parameters for this ATen function are non-trivially provided.  See the
+-- `Typed.NN.LSTM` module for doctests.
+--
+lstm
+  :: forall
+       numLayers
+       numDirections
+       dtype
+       seqLen
+       batchSize
+       inputDim
+       hiddenSize
+       device
+   . (KnownNat numLayers, KnownNat numDirections, KnownNat hiddenSize)
+  => Tensor device dtype '[seqLen, batchSize, inputDim]
+  -> ( Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     , Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     )
+  -> [D.Tensor]
+  -> Double
+  -> Bool
+  -> ( Tensor
+         device
+         dtype
+         '[seqLen, batchSize, hiddenSize * numDirections]
+     , Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     , Tensor
+         device
+         dtype
+         '[numLayers * numDirections, batchSize, hiddenSize]
+     )
+lstm _input (_cc, _hc) _params _dropout _train =
+  unsafePerformIO $ (cast9 ATen.lstm_tllbldbbb) _input
+                                                _hx
+                                                _params
+                                                True -- _has_biases
+                                                _num_layers
+                                                _dropout
+                                                _train
+                                                _bidirectional
+                                                False -- _batch_first
+ where
+  _hx =
+    [_cc, _hc] :: [ Tensor
+          device
+          dtype
+          '[numLayers * numDirections, batchSize, hiddenSize]
+      ]
+  (_num_layers :: I.Int64) =
+    fromIntegral $ natVal (undefined :: Proxy numLayers)
+  _bidirectional = case natVal (undefined :: Proxy numDirections) of
+    1 -> False
+    2 -> True
+    _ -> error "lstm: numDirections must be 1 or 2!"
+
+-- gru_cell :: Tensor dtype shape -> Tensor dtype shape -> Tensor dtype shape -> Tensor dtype shape -> Tensor dtype shape -> Tensor dtype shape -> Tensor dtype shape
```

Comment:
could you add the `lstmCell` doc and doctests back here please?

Reply:
I though I accepted both changes but it looks like the merge ate it.  It's been restored.